Fixed screw and screw hole

mgesteiro opened the pull request:
Hi Arne!

I like your design/puzle (found it [here](https://youtu.be/sPrPzY1m0aY?feature=shared&t=227))!

After printing it, I discovered a problem with the screw hole in the box: it was too short and I couldn't screw the lid all the way.

After (quite) some tinkering I managed to solve it and here it is my pull-request proposal with some changes:

1. Re-positioned the threaded rod in the lid (it's now fully embedded).
2. Fixed the screw hole in the box (there is now enough room to accept the whole lid screw).
3. Added pre-rendered STL files (they are quite difficult to generate even with a high-end computer, and also OpenScad is not _beginners friendly_).
4. Minor changes in the README, including a mention to BOSL2 library dependency and the STL files.

Please use these suggestions at your own discretion, I'm only trying to help 😊

Best regards!

Commit:
Fixed screw and screw hole, mention to the BOSL2 library dependency and prerendered STLs

akoehn commented:
Hey, sorry you had to deal with openscad code lol.  Thanks for the video link, I was completely unaware.

The compiled files are actually [on printables](https://www.printables.com/model/417667-bent-packing), so no need to compile them yourself or to commit them here.

I printed it again for a company gathering and might have not committed the newest openscad code; there was an intermediate state where the screw was incorrect as you described.

It is a bit late today but I will have a look later. Thanks for taking care and I hope you have fun with the puzzle :)

akoehn commented:
Oh, P.S. with the right backend, the whole set of pieces render in 3 seconds on my laptop -- openscad has made huge improvements in that regard.

akoehn commented:
There were indeed changes I forgot to commit, fixing exactly what you wrote about. That is now committed. I also added a link to printables into the README.

I incorporated your README improvements in de882a28e4b29032e01f4cfdb0bf23d91219a3dc. That was easier then modifying this PR.

I checked and rendering the whole build plate takes 34 seconds on my pretty normal laptop, so in case you continue to work with openscad, consider using the manifold backend, the old one is slow indeed with minkowski operations.

Thanks again for this PR and all the love to details, very much appreciated!

mgesteiro commented:
Hello again!

> There were indeed changes I forgot to commit, fixing exactly what you wrote about. That is now committed. I also added a link to printables into the README.

Nice 😊
 
> I incorporated your README improvements in [de882a2](https://github.com/akoehn/bent_packing/commit/de882a28e4b29032e01f4cfdb0bf23d91219a3dc). That was easier then modifying this PR.

Sure: I'm more than ok with the changes. Thanks for taking my PR into account.
 
> I checked and rendering the whole build plate takes 34 seconds on my pretty normal laptop, so in case you continue to work with openscad, consider using the manifold backend, the old one is slow indeed with minkowski operations.

I'll look into it (I suppose that it's not included in the default download and only in the nightly build 🤔).
I rarely use OpenScad, I'm more a FreeCAD rat: maybe I'll convert your design into it (those hexagons seem like a nice modelling challenge).

> Thanks again for this PR and all the love to details, very much appreciated!

You're welcome! It was nice of you to use my suggestions.

Best regards!
